### DISCUSSION TITLE
XMobar keeps showing "Updating..." for volume, brightness and power widgets

### DISCUSSION THREAD

**Discussion by aholmes2534** [antoniosarosi/dotfiles#21] [Q&A] [answered]
When I run your XMonad config using Xephyr, everything works perfectly, but when I log in via LightDM and actually use it, it only displays "Updating" for the battery, brightness and volume widgets. I tried searching for a way out of this but have spent a few hours and have failed to find any errors. ( The scripts work perfectly when invoked via the terminal, and I've added .local/bin to PATH and installed all the dependencies too. ) I'd really appreciate some help with this...

Thanks!

**Comment by antoniosarosi** [comment, marked as answer]
How did you add `.local/bin` to `$PATH`? If you didn't use the full route it won't work:
```bash
export PATH=$HOME/.local/bin:$PATH
```
If the scripts work from a terminal when invoking them *without* the full path, like `battery` instead of `~/.local/bin/battery`, then that's not the problem. Check if the file where `$PATH` is exported is actually executed when you login. If you do that in `~/.xprofile` add a line similar to this one: `touch ~/text.txt`. If the file is created then that's not the problem either.

Last thing that comes to my mind right now is that in the xmobar config there's a line that uses a full route containing my user name.
[`.config/xmobar/primary.hs`](https://github.com/antoniosarosi/dotfiles/blob/master/.config/xmobar/primary.hs):
```haskell
Run Com "/home/antonio/.config/xmobar/trayer-padding-icon.sh" [] "trayerpad" 600,
```
If you haven't already, change the username in that path to yours. It doesn't work with `$HOME` or `~`.

**Reply by aholmes2534** [reply]
Thank you so much for your patient reply! While my path declaration was right, seems like I put it in the wrong place... It's working now. 👍🏾
